### PR TITLE
Alh/sequences

### PR DESCRIPTION
**Pull request by wolfsage**

### CODE REVIEW

File: lib/Ix/DBIC/Schema.pm [hunk 1]
```diff
@@ -7,56 +7,144 @@ use parent 'DBIx::Class';
...
-   i := i + 1;
- END LOOP;
- RETURN ((r1 << 16) + l1);
-END;
-$$ LANGUAGE plpgsql strict immutable;
+  IF (octet_length(cr_key)!=10) THEN
+    RAISE EXCEPTION 'The encryption key must be exactly 10 bytes long.';
+  END IF;
+
+  IF (val > (2^32)-1 OR val < 0) THEN
```
Comment (wolfsage) [inline lib/Ix/DBIC/Schema.pm]: Added lines 46-48

File: lib/Ix/DBIC/Schema.pm [hunk 2]
```diff
@@ -7,56 +7,144 @@ use parent 'DBIx::Class';
...
+    RAISE EXCEPTION 'We can only handle values in the range 0 - (2^32)-1';
+  END IF;
+
```
Comment (wolfsage) [inline lib/Ix/DBIC/Schema.pm:49]: added 46-48 above

File: lib/Ix/DBIC/Schema.pm [hunk 3]
```diff
@@ -7,56 +7,144 @@ use parent 'DBIx::Class';
...
+    -- we've changed it to only deal in positive values. To make this work
+    -- we must translate any values over the max positive value ((2^31)-1)
+    -- into their negative counterparts
+    -- 2^31   becomes -1
+    -- 2^31+1 becomes -2, etc...
+
+    val := val - (2^31-1);
+    val := val * -1;
+  END IF;
+
```
Comment (wolfsage) [inline lib/Ix/DBIC/Schema.pm:69]: added 58-68 above

File: lib/Ix/DBIC/Schema.pm [hunk 4]
```diff
@@ -7,56 +7,144 @@ use parent 'DBIx::Class';
...
+  if (ret < 0) THEN
+    -- Like on input, we only want consumers to see positive values, so
+    -- we must translate negative values to their positive counterparts
+    -- -1 becomes 2^31
+    -- -2 becomes 2^31+1, etc...
+    ret := ret * -1;
+    ret := ret + (2^31-1); 
+  END IF;
+
```
Comment (wolfsage) [inline lib/Ix/DBIC/Schema.pm:103]: added 95-102 above

**Comment by rjbs** [comment]
I believe this is all correct, at least the bits that I didn't skip because they were copy-pasted crypotgraphy ;)

What else is left here?

**Comment by rjbs** [comment]
+1 on the idstr changes and in general

**Comment by rjbs** [comment]
I believe this is obsoleted by #57.

**Comment by wolfsage** [comment]
Correct, thanks

**Pull request closed without merging**